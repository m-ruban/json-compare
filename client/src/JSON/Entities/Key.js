import React from 'react'
import {Badge} from 'reactstrap'

const Key = ({value, isArray}) => {

  if (isArray){
    return (
      <Badge color="number">{value}</Badge>
    )
  }

  return (
    <>"{value}"</>
  )

}

export default Key
