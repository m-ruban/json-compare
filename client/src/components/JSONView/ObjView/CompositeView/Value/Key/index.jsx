import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Mark from 'components/Mark';

const Key = ({ value, isArray }) => {
    if (isArray) {
        return <Mark type="number" text={value} />;
    }

    return <Fragment>&quot;{value}&quot;</Fragment>;
};

Key.propTypes = {
    value: PropTypes.string,
    isArray: PropTypes.bool,
};

export default Key;
