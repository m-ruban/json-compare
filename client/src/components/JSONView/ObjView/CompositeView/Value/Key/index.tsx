import React from 'react';

import Mark from 'components/Mark';

const Key = ({ value, isArray }: { value: string; isArray: boolean; }) => {
    if (isArray) {
        return <Mark type="number" text={value} />;
    }
    return <span>{'"'}{value}{'"'}</span>;
};

export default Key;
