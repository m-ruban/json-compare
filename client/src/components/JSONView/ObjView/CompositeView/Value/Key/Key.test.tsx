import React from 'react';

import Key from 'components/JSONView/ObjView/CompositeView/Value/Key';
import { render } from 'enzyme';

describe('<Key />', () => {
    it('Renders correctly', () => {
        const props = { value: 'test', isArray: false };
        const output = render(<Key {...props} />);
        expect(output).toMatchSnapshot();
    });
});
