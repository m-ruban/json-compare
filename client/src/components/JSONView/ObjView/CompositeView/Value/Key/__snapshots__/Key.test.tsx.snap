// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Key /> Renders correctly 1`] = `
<span>
  "test"
</span>
`;
